src/Request/MultiPartFormDataRequest.php

```diff
@@ -22,7 +22,112 @@ class MultiPartFormDataRequest extends WebRequest
 
     public function getPayload()
     {
+        // Support multipart file PUT operations
+        if (isset($this->serverData['REQUEST_METHOD']) && $this->serverData['REQUEST_METHOD'] === 'PUT') {
+            $this->parsePut();
+        }
+
         $requestBody = array_merge($_FILES, $_POST);
         return $requestBody;
     }
+
+    /**
+     * Parses PUT request input into $_FILES
+     * @link http://stackoverflow.com/a/18678678
+     */
+    private function parsePut()
+    {
+        global $_PUT;
+
+        /* PUT data comes in on the stdin stream */
+        $putStream = fopen("php://input", "r");
+
+        $rawData = '';
+
+        /* Read the data 1 KB at a time
+           and write to the file */
+        while ($chunk = fread($putStream, 1024)) {
+            $rawData .= $chunk;
+        }
+
+        $rawData = ltrim($rawData);
+
+        /* Close the streams */
+        fclose($putStream);
+
+        // Fetch content and determine boundary
+        $boundary = substr($rawData, 0, strpos($rawData, "\r\n"));
```

Comment:
@samnotsowise There is some duplication here - take a look at MimeDocument::fromString() - it would take this string and return a tree of parts objects. The use of objects might make it slightly slower than this direct approach - it was originally designed for handling emails and mht files - but it would avoid duplication of code perhaps?